satpy/writers/cf_writer.py

```diff
@@ -685,14 +685,73 @@ def _cleanup_attrs(new_data):
 
     @staticmethod
     def _encode_coords(new_data):
+        """Encode coordinates."""
+        if not new_data.coords.keys() & {"x", "y", "crs"}:
```

Comment:
A KeysView from `.keys()` works with set logic? Cool! I'm also kind of surprised by the order of operations, but if it works then :+1:

Reply:
I tried it without the `.keys()` call first and that failed, but yes, `KeysView`s can do set logic :)